app/uk/gov/hmrc/cgtpropertydisposals/repositories/returns/DraftReturnsRepository.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposals.repositories.returns
+
+import cats.data.EitherT
+import com.google.inject.{ImplementedBy, Inject, Singleton}
+import play.api.libs.json._
+import cats.syntax.either._
+import cats.instances.list._
+import cats.syntax.traverse._
+import cats.instances.either._
+import uk.gov.hmrc.cgtpropertydisposals.util.JsErrorOps._
+import scala.concurrent.Future
+import play.modules.reactivemongo.ReactiveMongoComponent
+import reactivemongo.api.Cursor
+import reactivemongo.bson.BSONObjectID
+import uk.gov.hmrc.cgtpropertydisposals.config.AppConfig
+import uk.gov.hmrc.cgtpropertydisposals.models.Error
+import uk.gov.hmrc.cgtpropertydisposals.models.returns.DraftReturn
+import uk.gov.hmrc.cgtpropertydisposals.repositories.CacheRepository
+import uk.gov.hmrc.mongo.ReactiveRepository
+import reactivemongo.play.json.ImplicitBSONHandlers.JsObjectDocumentWriter
+import scala.concurrent.ExecutionContext.Implicits.global
+
+@ImplementedBy(classOf[DefaultDraftReturnsRepository])
+trait DraftReturnsRepository {
+  def fetch(cgtReference: String): EitherT[Future, Error, List[DraftReturn]]
```

Comment:
could we make `cgtReference` better typed? We have a `CgtReference` case class for that.